nova/virt/vmwareapi/host.py

```diff
@@ -124,10 +124,10 @@ def update_status(self):
             "numa_topology": None,
         }
 
-        for host, stats in per_host_stats.items():
-            data[host] = self._merge_stats(host, stats, defaults)
-
-        cluster_stats = vm_util.aggregate_stats_from_cluster(per_host_stats)
+        for host_ref, host in hosts_with_stats.items():
+            data[host["name"]] = self._merge_stats(host, host["stats"],
```

Comment:
I think the first argument to `_merge_stats()` needs to be `host['name']` instead.